components/Text/NumericValue.tsx

```diff
@@ -0,0 +1,36 @@
+import { wei, WeiSource } from '@synthetixio/wei';
+import { FC, memo, useMemo } from 'react';
+import styled from 'styled-components';
+
+import Body, { BodyProps } from './Body';
+
+type NumericValueProps = BodyProps & {
+	value: WeiSource;
+	preview?: boolean;
```

Comment:
Does this mean to not color it? I think we should go the other way and add a prop that opts in to color. Also there are labels where they can be positive but we show a negative color because the direction changed, I think Troy created a component for this and not sure if we want to combine here in some way.

Reply:
Or alternatively we have a base of `NumericValue` and an extended component like `ColoredNumericValue`

Reply:
The `preview` prop handles the gold (or yellow, in new designs) color that we use for numeric data when generating previews. See the screenshot below for reference:
<img width="296" alt="image" src="https://user-images.githubusercontent.com/15985212/220184877-5560ff2e-6bdd-4d11-9cc3-8560b3211b70.png">

Reply:
That said, you are right. I should definitely add a prop that enables/disables the colored text feature. By default, the `NumericValue` component will be equivalent to `<Text.Body mono />`, but with enhance color features based on the props.

Reply:
Yea I think it would be nice to be able to tell it to color based on value or color it ourself e.g. Troys `ColoredPrice` would have this as its child and color manually. Also wondering if this component is too much of a base level component for something like a `preview` prop which is quite specific to only trade previews in futures right now.

Reply:
Added the ability for this to be either colored or not in one of the newer commits. To your other point, I don't think it's too base-level to add the preview prop, since that seems to be the only other color used on numeric values (besides red/green and neutral colors).